module/zfs/zvol.c

```diff
@@ -65,7 +65,17 @@ static list_t zvol_state_list;
 #define	ZVOL_HT_SIZE	1024
 static struct hlist_head *zvol_htable;
 #define	ZVOL_HT_HEAD(hash)	(&zvol_htable[(hash) & (ZVOL_HT_SIZE-1)])
-static DEFINE_IDA(zvol_ida);
+
+/*
+ * Note: this is an unrolled version of the DEFINE_IDA() macro,
+ * modified to not use a cast.  The cast does not compile with the
+ * compiler flags used for ZoL (-std=gnu99).
```

Comment:
It's tripping up on the cast of a literal struct (casting it to the expected type, which is unnecessary but I wouldn't think would cause a problem).  But yeah, potentially a gcc bug.